types/src/key.rs

```diff
@@ -511,6 +511,14 @@ impl Key {
         hasher.finalize_variable(|hash| addr.clone_from_slice(hash));
         Key::Dictionary(addr)
     }
+
+    /// Returns true if the key is of type [`Key::Dictionary`].
+    pub fn is_dictionary_key(&self) -> bool {
```

Comment:
Shouldn't need this if we'd make new host function accept a DictionaryAddr rather than arbitrary Key.